src/main/java/cloud/eppo/BaseEppoClient.java

```diff
@@ -22,37 +22,46 @@ public class BaseEppoClient {
   private final ObjectMapper mapper =
       new ObjectMapper()
           .registerModule(EppoModule.eppoModule()); // TODO: is this the best place for this?
-  private static final String DEFAULT_HOST = "https://fscdn.eppo.cloud";
-  private static final boolean DEFAULT_IS_GRACEFUL_MODE = true;
+
+  protected static final String DEFAULT_HOST = "https://fscdn.eppo.cloud";
+  protected final ConfigurationRequestor requestor;
 
   private final ConfigurationStore configurationStore;
-  private final ConfigurationRequestor requestor;
   private final AssignmentLogger assignmentLogger;
   private final BanditLogger banditLogger;
   private final String sdkName;
   private final String sdkVersion;
   private final boolean isConfigObfuscated;
   private boolean isGracefulMode;
 
-  private static BaseEppoClient instance;
-
   // Fields useful for testing in situations where we want to mock the http client or configuration
   // store (accessed via reflection)
   /** @noinspection FieldMayBeFinal */
   private static EppoHttpClient httpClientOverride = null;
 
-  private BaseEppoClient(
+  public BaseEppoClient(
       String apiKey,
       String sdkName,
       String sdkVersion,
       String host,
-      ConfigurationStore configurationStore,
       AssignmentLogger assignmentLogger,
       BanditLogger banditLogger,
-      boolean isGracefulMode) {
+      boolean isGracefulMode,
+      boolean expectObfuscatedConfig) {
+
+    if (apiKey == null) {
+      throw new IllegalArgumentException("Unable to initialize Eppo SDK due to missing API key");
+    }
+    if (sdkName == null || sdkVersion == null) {
+      throw new IllegalArgumentException(
+          "Unable to initialize Eppo SDK due to missing SDK name or version");
+    }
+    if (host == null) {
+      host = DEFAULT_HOST;
+    }
 
     EppoHttpClient httpClient = buildHttpClient(host, apiKey, sdkName, sdkVersion);
-    this.configurationStore = configurationStore;
+    this.configurationStore = new ConfigurationStore();
```

Comment:
we were always just passing in `new ConfigurationStore()` so in-lining this for now until we handle caching at which point I reckon this may find its way back to the constructor, but didn't want to put the cart before the horse.